overlord/snapstate/snapmgr.go

```diff
@@ -428,6 +458,17 @@ func (m *SnapManager) doMountSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// record the try mode in the state
+	t.State().Lock()
+	t.Set("old-trymode", snapst.TryMode())
+	if ss.TryMode() {
+		snapst.Flags |= TryMode
+	} else {
+		snapst.Flags &= ^TryMode
+	}
+	Set(t.State(), ss.Name, snapst)
+	t.State().Unlock()
```

Comment:
this saves trymode before we know if SetupSnap failed or not, so on SetupSnap error it will stay there, is that what we want?